### PR TITLE
Allow using an alternate suite repo

### PR DESCRIPTION
**Pull request by zmc**
... and also optionally look for tasks and suite definitions in a subdirectory of that repo

Related:
http://pad.ceph.com/p/qa-suite-merge
https://paste.fedoraproject.org/493541/04623891/

### CODE REVIEW

**Comment by zmc** [comment]
Note to self: repos on-disk need to be named to avoid collisions

File: teuthology/suite/run.py [hunk 1]
```diff
@@ -189,30 +194,42 @@ def choose_teuthology_branch(self):
...
+            return 'ceph-qa-suite'
+
     def choose_suite_branch(self):
+        suite_repo_name = self.suite_repo_name
         suite_branch = self.args.suite_branch
         ceph_branch = self.args.ceph_branch
         if suite_branch and suite_branch != 'master':
-            if not util.git_branch_exists('ceph-qa-suite', suite_branch):
-                exc = BranchNotFoundError(suite_branch, 'ceph-qa-suite.git')
+            if not util.git_branch_exists(suite_repo_name, suite_branch):
```
Comment (liewegas) [inline teuthology/suite/run.py]: Does this assume that it's a repo under github.com/ceph?

Comment (zmc) [inline teuthology/suite/run.py]: Ah, it will but it doesn't have to be that way of course. I'll work on that in the morning

**Comment by zmc** [comment]
@liewegas I *believe* this does what we want now!

**Comment by zmc** [comment]
Testing new worker code with new suite features:
http://pulpito.ceph.com/zack-2016-12-02_09:49:30-smoke-master---basic-mira/596203/

Testing new worker code without new suite features:
http://pulpito.ceph.com/zack-2016-12-02_09:50:46-smoke-master---basic-mira/596205/

Both are doing what I would expect.

**Comment by zmc** [comment]
I'm going to push some unit-test-only changes

**Comment by dmick** [comment]
Do you want to wait for those unit test changes?

**Comment by zmc** [comment]
@dmick I pushed them Friday morning

**Comment by dmick** [comment]
so confused; gh doesn't show any new commits after your comment.  sigh.

**Comment by zmc** [comment]
It was a force push on top of an existing commit

File: teuthology/repo_utils.py [hunk 2]
```diff
@@ -97,20 +97,25 @@ def enforce_repo_state(repo_url, dest_path, branch, remove_on_error=True):
         raise
 
 
-def clone_repo(repo_url, dest_path, branch):
+def clone_repo(repo_url, dest_path, branch, shallow=True):
```
Comment (zmc) [inline teuthology/repo_utils.py]: Are you sure? I couldn't find any.

Comment (dmick) [inline teuthology/repo_utils.py]: It's all inside repo_utils, but it gets used for teuthology and ceph-qa-suite both.  The other one I saw was in the cephfs tests in ceph-qa-suite, but I see now that's a locally-defined routine of the same name.

Mostly I was raising the question to make certain you'd doublechecked all the call paths.  I can't think of why teuthology would care that it's depth 1, but it's a change.

Comment (zmc) [inline teuthology/repo_utils.py]: So yeah, there's one caller of `clone_repo` - it's the one that needs the changes here.

I can't imagine why a shallow clone would disturb anything else teuthology does. It's eternally possible that I've missed something, though :)

Comment (dmick) [inline teuthology/repo_utils.py]: yeah.  and nothing appears to use fetch_branch but tests.  shrug.  It's likely fine.

**Comment by liewegas** [comment]
a shallow clone also seems safe to me.  +1

**Pull request merged**